lib/engine/game/g_1862/game.rb

```diff
@@ -1196,7 +1196,7 @@ def sell_shares_and_change_price(bundle, allow_president_change: true, swap: nil
 
           @share_pool.sell_shares(bundle, allow_president_change: allow_president_change, swap: swap)
           num_shares = bundle.num_shares
-          unless corporation.owner == bundle.owner
+          unless corporation.owner == bundle.president
```

Comment:
What am I missing? Isn't bundle.president defined to be corporation.owner? Won't this always be true?

Reply:
bundle.owner points to the sharepool object that the bundle is in, afaik

Reply:
My point is that bundle.president always points to corporation.owner, regardless of who owns the bundle. This means that the price will always drop even if it's not the president selling the shares.

The bug is that I don't make this check until after calling sell_shares, and thus the price never drops for one or two shares.

Reply:
Ah I see what you're saying

Further context to my above comment which I think was part of my confusion, `bundle.owner` in the bugged implementation returned a `SharePool` object after `sell_shares`, so the comparison was silently failing

https://github.com/tobymao/18xx/issues/8916